lib/std/Thread.zig

```diff
@@ -199,7 +199,9 @@ pub fn spawn(comptime startFn: anytype, context: SpawnContextType(@TypeOf(startF
                 inner: Context,
             };
             fn threadMain(raw_arg: windows.LPVOID) callconv(.C) windows.DWORD {
-                const arg = if (@sizeOf(Context) == 0) {} else @ptrCast(*Context, @alignCast(@alignOf(Context), raw_arg)).*;
+                const arg = if (Context == void) {} //
+                else if (@sizeOf(Context) == 0) .{} //
```

Comment:
This works for 0 size structs but not for `u0`, enums with 1 tag, etc. Seems like there should be a simpler fix that only checks if `@sizeOf` is 0.

Reply:
Ah, good catch. Let's use `undefined` instead? That should work for every zero-sized type.

Reply:
Sounds good